xla/service/gpu/cudnn_fused_mha_rewriter.cc

```diff
@@ -59,6 +65,49 @@ bool IsBatchedMatmul(const HloInstruction* instr) {
   return is_batch_dot;
 }
 
+// We need to check if current gemm is sharing a parent node with a forward
+// fMHA call. We check this by doing a BFS of all operands to see if there's
+// any user that is a forward fMHA custom call.
+bool IsSharingOperandWithFwdMha(HloInstruction* gemm) {
+  for (int64_t i = 0; i < gemm->operands().size(); i++) {
+    std::queue<HloInstruction*> visit_list;
+    visit_list.push(gemm->mutable_operand(i));
+    while (!visit_list.empty()) {
```

Comment:
Could you add a comment describing the traversal logic a bit more?

Reply:
added more comments